American Whitewater/ViewControllers/Filtering/FilterViewController.swift

```diff
@@ -0,0 +1,516 @@
+import UIKit
+import CoreLocation
+
+class FilterViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, UISearchBarDelegate, CLLocationManagerDelegate,
+                            UICollectionViewDelegate, UICollectionViewDelegateFlowLayout, UICollectionViewDataSource {
+    
+    @IBOutlet weak var contentCollectionView: UICollectionView!
+    
+    @IBOutlet weak var filterSegmentControl: UISegmentedControl!
+       
+    let locationManager = CLLocationManager()
+    let geoCoder = CLGeocoder()
+    
+    var selectedRegions:[Region] = []
+    var selectedRegionsOrig:[Region] = []
+    var usaRegions = Region.states
+    var internationalRegions = Region.international
+    var classFilters:[Int] = DefaultsManager.classFilter
+
+    @IBOutlet weak var regionsContainerView: UIView!
+        
+    var regionsTableViewRef: UITableView?
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+     
+        
+        filterSegmentControl.fallBackToPreIOS13Layout(using: UIColor.white)
+        let selectedSegTitle = [NSAttributedString.Key.foregroundColor: UIColor(named: "primary") ?? UIColor.black]
+                                as [NSAttributedString.Key : Any]
+        filterSegmentControl.setTitleTextAttributes(selectedSegTitle, for: .selected)
+
+    }
+
+    func imageWithColor(color: UIColor) -> UIImage? {
+        
+        let rect = CGRect(origin: .zero, size: CGSize(width: 1, height: 1))
+        UIGraphicsBeginImageContextWithOptions(rect.size, false, 0.0)
+        color.setFill()
+        
+        let image = UIGraphicsGetImageFromCurrentImageContext()
+        
+        UIGraphicsEndImageContext()
+
+        guard let img = image else { return nil }
+
+        return img
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        loadRegions()
+
+        contentCollectionView.reloadData()
+        
+    }
+    
+    
+    
+    // Filter Type Changed changes the filter category (region/class/distance)
+    // and auto scrolls to the correct view for interaction
+    @IBAction func filterTypeChanged(_ segmentControl: UISegmentedControl) {
+        
+        contentCollectionView.scrollToItem(at: IndexPath(item: segmentControl.selectedSegmentIndex, section: 0), at: .left, animated: true)
+        
+    }
+    
+    
+    //
+    // MARK - Regional Filtering
+    //
+    
+    
+    // Clears regions from search and local stroage
+    @objc func clearReagionsButtonPressed(_ sender: Any) {
```

Comment:
Minor: typo. Also seeing a lot of typos in the comments, but not bothering to comment on those.

Reply:
Ya, xcode's spell check must have turned off during an upgrade. I'm not surprised at how bad this was :) It's resolved now but I'm sure I missed 1 or 2. I'll keep an eye out for them.